mungegithub/mungers/publish_scripts/util.sh

```diff
@@ -38,22 +38,22 @@ sync_repo() {
 
     local currBranch=$(git rev-parse --abbrev-ref HEAD)
     local previousKubeSHA=$(cat kubernetes-sha)
-    local previousBranchSHA=$(cat filter-branch-sha)
     
     git remote add upstream-kube https://github.com/kubernetes/kubernetes.git || true
     git fetch upstream-kube
     git branch -D kube-sync || true
     git checkout upstream-kube/"${src_branch}" -b kube-sync
     git reset --hard upstream-kube/"${src_branch}"
-    local newKubeSHA=$(git log --oneline --format='%H' kube-sync -1)
     
     # this command rewrites git history to *only* include $subdirectory 
-    git filter-branch -f --subdirectory-filter "${subdirectory}" HEAD
-    
-    newBranchSHA=$(git log --oneline --format='%H' kube-sync -1)
-    # we need to start after _vendor is removed from master
+    git filter-branch -f --msg-filter 'awk 1 && echo && echo "Kubernetes-commit: ${GIT_COMMIT}"' \
+        --subdirectory-filter "${subdirectory}" HEAD
+
+    local newKubeSHA=$(git log kube-sync -1 | tail -n 1 | sed "s/Kubernetes-commit: //g")
+
+    local previousBranchSHA=$(git log --grep "Kubernetes-commit: ${previousKubeSHA}" --format='%H')
```

Comment:
Didn't consider the merge commits, you are right.

What I meant with "nice and clean" is that the client-go /apimachienry history will look differently than kube's. No idea whether it is feasible to make those "sync commits" look like those merge commits, just an idea....